tapdb/assets_store.go

```diff
@@ -883,12 +883,40 @@ func (a *AssetStore) constraintsToDbFilter(
 				Valid: true,
 			}
 		}
+
+		if query.MaxAmt != 0 {
+			assetFilter.MaxAmt = sql.NullInt64{
+				Int64: int64(query.MaxAmt),
+				Valid: true,
+			}
+		} else {
+			// If MaxAmt is unspecified, use valid: false to ignore
+			// the amount option in the SQL query
+			assetFilter.MaxAmt = sql.NullInt64{
+				Valid: false,
+			}
+		}
+
 		if query.MinAnchorHeight != 0 {
 			assetFilter.MinAnchorHeight = sqlInt32(
 				query.MinAnchorHeight,
 			)
 		}
 
+		if query.ScriptKeyID != nil {
```

Comment:
we can't really have user-facing arguments that reference database IDs

the `ScriptKeyID` is the DB identifier for a script key and there's no possible way a user could (...easily) get their hands on one of them in order to filter the assets

Instead we could have `*AssetQueryFilters` expose `scriptKey []byte` and `anchorOutpoint []byte` filters, which the user can easily acquire, then on this point in the code we could map these values to the unique db ID that the underlying layer requires for filtering (that is `QueryAssetFilters` struct)

Reply:
if you add some tests with the current way things work, you'll see that when you'll try to query a specific script key you'll have trouble finding the ID to filter by